### DISCUSSION TITLE
Some Issues on Event Trigerred

### DISCUSSION THREAD

**Discussion by Akiitp** [eopeter/flutter_mapbox_navigation#262] [Q&A]
Thanks, for making very nice library. 
we Faceing Some Issues. MapBoxEvent.banner_instructions is never triggered
Future<void> _onEmbeddedRouteEvent(e) async {
    _distanceRemaining = await MapBoxNavigation.instance.getDistanceRemaining();
    _durationRemaining = await MapBoxNavigation.instance.getDurationRemaining();

    switch (e.eventType) {
      case MapBoxEvent.progress_change:
        var progressEvent = e.data as RouteProgressEvent;
        if (progressEvent.currentStepInstruction != null) {
          _instruction = progressEvent.currentStepInstruction;
          // _currentStepIndex = progressEvent.;
          var currentStep = progressEvent.currentLeg!.steps![3].distance;
          print("currentStepIndex: ${currentStep}");
          if (!isArrived) {
            _sendMapData(_instruction, _distanceRemaining);
          }
          else {
            if (_controller != null) {
              await _controller?.finishNavigation();
            }
          }
        }
        break;

      case MapBoxEvent.route_building:
      case MapBoxEvent.route_built:
        setState(() {
          _routeBuilt = true;
        });
        break;

      case MapBoxEvent.route_build_failed:
        setState(() {
          _routeBuilt = false;
        });
        break;

      case MapBoxEvent.navigation_running:
        setState(() {
          _isNavigating = true;
        });
        break;

      case MapBoxEvent.on_arrival:
        setState(() {
          isArrived = true;
        });
        if (_controller != null) {
          await _controller?.finishNavigation();
        }
        break;

      case MapBoxEvent.navigation_finished:
      case MapBoxEvent.navigation_cancelled:
        setState(() {
          _routeBuilt = false;
          _isNavigating = false;
          isArrived = false;
        });
        break;
      case MapBoxEvent.banner_instruction:
        var banner_instructions=e.data;
        print("banner_instructions:  ${banner_instructions}");
        break;

      default:
        break;
    }

    setState(() {});
  }

**Comment by eopeter** [comment]
Seems the data you are looking for is on the progress_changed event

**Reply by Akiitp** [reply]
Actually I want  try to get modifers value and type

**Reply by Akiitp** [reply]
In Current Leg Step is Value Always getting Null Value and remaings Legs Value always getting Null Value.
I also Check your Sample App.

**Comment by eopeter** [comment]
So looks like a bug that needs to be worked on. Will take a look when I have some time.

**Reply by Akiitp** [reply]
I Thinks bugs on  Json Parsing that's  why we receive null Value. Inside RouteStep Class inSide this Class Only We receive Distance and other any value getting Null

**Comment by eopeter** [comment]
Is this happening on Android or iOS or both?

**Reply by Akiitp** [reply]
Currently I Check On only Android.
Issues in RouteLeg Class in this Class only Distance Value Get Proper . and Other Value Gettings Always null Value.

**Reply by Akiitp** [reply]
Any update of this Issues

**Reply by Akiitp** [reply]
Plz Update regarding issues
